cachito/workers/pkg_manager.py

```diff
@@ -61,19 +61,27 @@ def resolve_gomod_deps(archive_path, request_id=None):
 
         _run_cmd(('go', 'mod', 'download'), run_params)
         go_list_output = _run_cmd(
-            ('go', 'list', '-m', '-f', '{{.Path}} {{.Version}}', 'all'), run_params)
+            ('go', 'list', '-m', '-f', '{{.Path}} {{.Version}} {{.Replace}}', 'all'), run_params)
 
         deps = []
         for line in go_list_output.splitlines():
-            parts = line.split(' ')
+            # If there is no "replace" directive used on the dependency, then the last column will
+            # be "<nil>"
+            parts = [part for part in line.split(' ') if part not in ('', '<nil>')]
             if len(parts) == 1:
                 # This is the application itself, not a dependency
                 continue
-            if len(parts) > 2:
-                log.warning('Unexpected go module output: %s', line)
-                continue
+            elif len(parts) == 4:
```

Comment:
Consider using `if` instead of `elif` here. This seems to create some sort of relationship with the previous `if` statement that I don't think is accurate.